cmd/podman/machine/info.go

```diff
@@ -101,7 +101,12 @@ func hostInfo() (*entities.MachineHostInfo, error) {
 	host.Arch = runtime.GOARCH
 	host.OS = runtime.GOOS
 
-	provider := GetSystemDefaultProvider()
+	cfg, err := config.ReadCustomConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	provider := GetSystemProvider(cfg.Machine.Provider)
```

Comment:
Note that read custom config is not the right things to do here IMO,
the custom config only reads the containers.conf in your $HOME. 

Thus is will be impossible for admins or distributors to specify a different global default in /etc or /use/share/.


The custom config is only required for the system connection settings because the are per user and written by us.

Reply:
The internals are not that documented. Will check the sources for a better alternative. Thank you for the feedback!

Reply:
@Luap99 could you provide some more input about why `config.ReadCustomConfig()` is not suitable here as I just moved these lines from 113-117 up the source file. Was it a bug before or is the current user scenario different from what it was used before?

Reply:
I figured it out, that there is `Default()` public func to fetch all merged. Will use this one and move into GetProvider also handling errors inside.